### PR TITLE
feat(limits): Add "branchConcurrentLimit" option

### LINKED ISSUE #7076: New option: branchConcurrentLimit
**What would you like Renovate to be able to do?**

Support the capability for separate branch and PR limits

**Did you already have any implementation ideas?**

At branch creation time, we check if branchConcurrentLimit and branchHourlyLimit are defined. If not, then they fall back to prConcurrentLimit and prHourlyLimit respectively. Enforce this limit at branch creation time.

Note: I'm not sure if it's possible to easily measure hourly limits for branches, because we don't have a "creation" timestamp - just a commit timestamp (which could also be an update instead of create). If it's not feasible, then we can maybe only enforce concurrent for now.

Blocked by #7075

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

Separate PR limits from branch limits and allow to configure both. By default, branch limit is inherited from PR limit.

## Context:

Ref #7136
Closes #7076, #7197 and #7239

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [x] Unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
Test repo: https://github.com/zharinov/renovate-test-branch-limit

File: lib/workers/repository/process/limits.spec.ts [hunk 1]
```diff
@@ -82,4 +89,39 @@ describe('workers/repository/process/limits', () => {
...
+      git.branchExists.mockReturnValueOnce(true);
+      const res = limits.getBranchesRemaining(config, [
+        { branchName: 'foo' },
+      ] as never);
+      expect(res).toEqual(19);
+    });
+    it('does not use prConcurrentLimit for explicit branchConcurrentLimit=0', () => {
+      config.branchConcurrentLimit = 0;
+      config.prConcurrentLimit = 20;
+      const res = limits.getBranchesRemaining(config, [] as never);
```
Comment (viceice) [inline lib/workers/repository/process/limits.spec.ts:114]:
```suggestion
      const res = limits.getBranchesRemaining(config, []);
```
Cast not needed.

File: lib/workers/repository/process/limits.ts [hunk 2]
```diff
@@ -84,3 +85,40 @@ export async function getPrsRemaining(
...
+    }
+  }
+  return 99;
+}
+
+export function getBranchesRemaining(
+  config: RenovateConfig,
+  branches: BranchConfig[]
+): number {
+  return getConcurrentBranchesRemaining(config, branches);
```
Comment (viceice) [inline lib/workers/repository/process/limits.ts:123]: Do we really need two functions?

**Review by viceice — COMMENTED** [review]
We should validate that branch limit is not less than pr limit?

File: lib/workers/repository/process/limits.ts (hunk 2 above)
Comment (zharinov) [reply lib/workers/repository/process/limits.ts:123]: The intention was (1) to be consistent with `getPrsRemaining()` function and (2) have single extension point for the case when hourly branch limits are added

**Review by viceice — APPROVED** [review]

**Pull request merged**